win32ss/user/winsrv/usersrv/shutdown.c

```diff
@@ -191,8 +191,17 @@ EndNowThreadProc(LPVOID Parameter)
     PNOTIFY_CONTEXT NotifyContext = (PNOTIFY_CONTEXT)Parameter;
     MSG Msg;
 
-    // SetThreadDesktop(NotifyContext->Desktop);
-    // SwitchDesktop(NotifyContext->Desktop);
+#if 0
+    SetThreadDesktop(NotifyContext->Desktop);
+    SwitchDesktop(NotifyContext->Desktop);
+#else
+    /* For now show the end task dialog in the active desktop */
+    NtUserSetInformationThread(NtCurrentThread(),
+                               UserThreadUseActiveDesktop,
+                               NULL,
+                               0);
+#endif
```

Comment:
Don't you need to "restore" the old thread desktop at the end of the function?

Reply:
The thread would exit anyway when the function returns so there is not much use in it.